### PR TITLE
chore: lint fixes

### PR DESCRIPTION
**Pull request by aljo242**
## 1. Summary

Fixes failing lint

## 2.Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## 3. Implementation details

Golangci-lint action was bumped to `v1.52.0` which started complaining about a bunch of code.  This PR bumps our version of `lint` and makes the according changes to fix the lint.

- unused parameters
- redundant if-else
- simplifying if-else statements
- remove `NewHandler` functions (not needed anymore).

Ones to check (simplified logic that should be verified):
- x/interchainstaking/types/redemptions.go
- x/interchainstaking/keeper/redemptions.go
- x/interchainstaking/types/rebalance.go

### CODE REVIEW

File: x/interchainstaking/handler.go [hunk 1]
```diff
@@ -10,28 +10,6 @@ import (
 	"github.com/ingenuity-build/quicksilver/x/interchainstaking/types"
 )
 
-// NewHandler returns a handler for interchainstaking module messages
```
Comment (joe-bowman) [inline x/interchainstaking/handler.go:13]: I thought this was still used for amino messages? Or is it deprecated entirely now?

File: x/interchainstaking/keeper/msg_server.go [hunk 2]
```diff
@@ -237,12 +237,12 @@ func (k msgServer) GovCloseChannel(goCtx context.Context, msg *types.MsgGovClose
 			)
 	}
 
-	_, cap, err := k.Keeper.IBCKeeper.ChannelKeeper.LookupModuleByChannel(ctx, msg.PortId, msg.ChannelId)
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server.go:240]: is `cap` a keyword? I dislike `c` as a variable name, it's somewhat ambiguous. Can we make it `capability` instead?

File: x/interchainstaking/module.go [hunk 3]
```diff
@@ -125,14 +125,14 @@ func (am AppModule) Name() string {
 
 // Route returns the capability module's message routing key.
 func (am AppModule) Route() sdk.Route {
-	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper))
```
Comment (joe-bowman) [inline x/interchainstaking/module.go:128]: is this equivalent?

**Review by joe-bowman — COMMENTED** [review]
Are we sure about the NewHandler + Route() removal? I thought this was still used for Amino (ledger) messages?

**Comment by joe-bowman** [comment]
This suggests to me that it is still relevant in 0.46? https://github.com/cosmos/cosmos-sdk/blob/5023392bd9718679668dd8e88a792a6defa63c00/types/module/module.go#LL294C3-L296C4

File: x/interchainstaking/handler.go (hunk 1 above)
Comment (aljo242) [reply x/interchainstaking/handler.go:13]: https://github.com/cosmos/cosmos-sdk/pull/9650

Added in `v0.46.0`

File: x/interchainstaking/keeper/msg_server.go (hunk 2 above)
Comment (aljo242) [reply x/interchainstaking/keeper/msg_server.go:240]: `cap` a keyword (for "capacity").

I'll modify to "capability"

File: x/interchainstaking/types/accounts.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/ingenuity-build/quicksilver/utils"
 )
 
-func NewICAAccount(addr string, portID string, baseDenom string) (*ICAAccount, error) {
```
[the remark this answers is not in the document]
Comment (aljo242) [reply x/interchainstaking/types/accounts.go:11]: https://github.com/ingenuity-build/quicksilver/pull/356/commits/37b9b18a2600955db5ab97de747b77f5cec9b62b

File: x/interchainstaking/handler.go (hunk 1 above)
Comment (joe-bowman) [reply x/interchainstaking/handler.go:13]: thank you :)

**Review by joe-bowman — DISMISSED** [review]
lgtm

**Review by joe-bowman — COMMENTED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
LGTM

**Review by joe-bowman — APPROVED** [review]
lgtm

**Pull request merged**